irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -72,55 +78,81 @@ namespace torii {
+        request->transactions()
+        | boost::adaptors::transformed(
+              [&](const auto &tx) { return transaction_factory_->build(tx); })
+        | boost::adaptors::filtered([&](const auto &result) {
+            return result.match(
+                [](const iroha::expected::Value<
+                    std::unique_ptr<shared_model::interface::Transaction>> &) {
+                  return true;
+                },
+                [&](const iroha::expected::Error<TransportFactoryType::Error>
+                        &error) {
+                  status_bus_->publish(status_factory_->makeStatelessFail(
+                      error.error.hash, error.error.error));
+                  return false;
+                });
+          })
+        | boost::adaptors::transformed([&](auto result) {
+            return std::move(
+                       boost::get<iroha::expected::ValueOf<decltype(result)>>(
+                           result))
+                .value;
+          }));
+  }
+
   grpc::Status CommandServiceTransportGrpc::ListTorii(
       grpc::ServerContext *context,
       const iroha::protocol::TxList *request,
       google::protobuf::Empty *response) {
-    auto tx_list_builder = shared_model::proto::TransportBuilder<
-        shared_model::interface::TransactionSequence,
-        shared_model::validation::DefaultUnsignedTransactionsValidator>();
-
-    tx_list_builder.build(*request).match(
-        [this](
-            iroha::expected::Value<shared_model::interface::TransactionSequence>
-                &tx_sequence) {
-          this->command_service_->handleTransactionList(tx_sequence.value);
-        },
-        [this, request](auto &error) {
-          auto &txs = request->transactions();
-          if (txs.empty()) {
-            log_->warn("Received no transactions. Skipping");
-            return;
-          }
-          using HashType = shared_model::crypto::Hash;
-
-          std::vector<HashType> hashes;
-          std::transform(
-              txs.begin(),
-              txs.end(),
-              std::back_inserter(hashes),
-              [](const auto &tx) {
-                return shared_model::crypto::DefaultHashProvider::makeHash(
-                    shared_model::proto::makeBlob(tx.payload()));
+    auto transactions = deserializeTransactions(request);
+
+    auto batches = batch_parser_->parseBatches(transactions);
+
+    for (auto &batch : batches) {
+      shared_model::interface::TransactionBatchFactory::createTransactionBatch(
+          batch,
+          shared_model::validation::DefaultUnsignedTransactionsValidator())
+          .match(
+              [&](iroha::expected::Value<
+                  shared_model::interface::TransactionBatch> &value) {
+                this->command_service_->handleTransactionBatch(
+                    std::make_shared<shared_model::interface::TransactionBatch>(
+                        std::move(value).value));
```

Comment:
Are the parentheses placed correctly here? You move not the value, but the initial variable?

Reply:
Yes, this expression is safer. https://oliora.github.io/2016/02/12/where-to-put-std-move.html